interview-questions.md

```diff
@@ -16,32 +16,61 @@ Researched answer:
 
 2. What is the difference between an array's index and its value?
 
-Your answer:
+Your answer: An array's index represents a particular position in that array denoted by a number from 0 to however many values the array has. Each value in the array can be accessed using the index. The value of the array are the array components themselves, separated by a comma.
 
-Researched answer:
+Researched answer: An array is an ordered list of elements or values. The array is ordered by index which indicates the position of the element within the array starting from 0. 
 
 3. Compare and contrast arrays and strings. Please include ways in which they are the same and ways in which they are different.
 
-Your answer:
+Your answer: 
+Similarities:
+-both are data types
+-can store both into variables
+-arrays and strings can both be iterated through
+-they both share properties and methods like .length and .slice() to name a couple
+
+Differences:
+-certain methods and properties they don't share like .reverse()
+-can't use higher order functions on strings
+-arrays can hold different values/data types
+-strings are a primitive data type and arrays are a different data type that I can't remember the name of :)
+-strings are immutable and arrays are not
 
 Researched answer:
+Similarities:
+-both are collections of elements that are stored sequentially in memory
+-elements can be accessed using index values
+-both support various operations like concatenation, slicing, iteration, and searching
+-both are passable
+
+Differences:
+-type of data array can hold vs string
+-strings are usually can't be changed after creation while arrays are usually mutable
+-strings are representated by quotes (" ") or apostrophes (' ') while arrays are usually defined by square brackets ([])
 
 4. STRETCH: What are the four pillars of object oriented programming?
 
-Researched answer:
+Your answer: 
+I have this in my notes from Codeup :) 
+1) Abstraction which is abstracting the complexity or simplifying
+2) Encapsulation which is when all related functions are organized under one class. This helps avoid long naming conventions too
+3) Inheritance which uses "extends" keyword to get access to the properties of another object all in the interest of not repeating oneself 
+4) Polymorphism which is anything being changed when the "inheritance" is happening
+
+Researched answer: Honestly what I'm finding on the internet about OOP loosely makes sense to me so I am finding it a little difficult to not just copy and paste my findings
```

Comment:
Totally understandable!  You did a great job extracting the basic def in your Codeup notes!  Woohoo!